internal/objects/crdrules.go

```diff
@@ -76,6 +82,45 @@ func (c *CRDLister) GetFQDNToHostruleMapping(fqdn string) (bool, string) {
 	return true, hostrule.(string)
 }
 
+func (c *CRDLister) GetFQDNToHostruleMappingV2(fqdn string) (bool, string) {
+	c.NSLock.Lock()
+	defer c.NSLock.Unlock()
+	// not exact fqdns
+	allFqdns := c.FqdnHostRuleCache.GetAllKeys()
+	returnHostrules := []string{}
+	for _, mFqdn := range allFqdns {
+		oktype, fqdnType := c.FqdnFqdnTypeCache.Get(mFqdn)
+		if !oktype || fqdnType == "" {
+			fqdnType = string(akov1alpha1.Exact)
+		}
+
+		if fqdnType == string(akov1alpha1.Exact) && mFqdn == fqdn {
+			if found, hostrule := c.FqdnHostRuleCache.Get(mFqdn); found {
+				returnHostrules = append(returnHostrules, hostrule.(string))
+				break
+			}
+		} else if fqdnType == string(akov1alpha1.Contains) && strings.Contains(fqdn, mFqdn) {
+			if found, hostrule := c.FqdnHostRuleCache.Get(mFqdn); found {
+				returnHostrules = append(returnHostrules, hostrule.(string))
+				break
+			}
+		} else if fqdnType == string(akov1alpha1.Wildcard) && strings.HasPrefix(mFqdn, "*") {
+			wildcardFqdn := strings.Split(mFqdn, "*")[1]
```

Comment:
what if the user just specified a "*" and nothing else. Does this code fail in that case?

Reply:
you're right yes, thanks.

Reply:
actually no, this is fine. https://go.dev/play/p/NNRDA0cexeT